guillotina/transactions.py

```diff
@@ -135,3 +167,6 @@ def adopt_objects(self, obs, txn):
                 self.request._txn = self.previous_txn
             if self.previous_write_setting is not None:
                 self.request._db_write_enabled = self.previous_write_setting
+
+        if self.request is not None:
```

Comment:
Did you mean to check this.execute_futures ?
Should we sheck if request.execute_futures exist ?

Reply:
You are correct! Thank you for catching that.

"this.execute_futures": I can tell you've been doing a lot of JavaScript lately :)